apps/system/js/bluetooth_v2.js

```diff
@@ -435,6 +439,52 @@ Bluetooth.prototype = {
   },
 
   /**
+   * Get adapter from bluetooth through promise interface.
+   *
+   * @public
+   * @return {Promise} Bluetooth Adapter
+   */
+  adapter: function bt__adapter() {
+    return new Promise((resolve, reject) => {
+      if (this._adapter !== null) {
+        resolve(this._adapter);
+      } else {
+        this.debug('No BT adapter retrieved');
+        reject();
+      }
+    });
+  },
+
+  /**
+   * Return device pairing result.
+   *
+   * @public
+   * @param {string} mac target device address
+   * @return {Promise} paired result
+   */
+  pair: function bt__pair(mac) {
+    return new Promise((resolve, reject) => {
+      this._adapter.pair(mac).then(() => {
+        resolve(mac);
+      }, () => {
+        reject();
+      });
+    });
+  },
+
+  /**
+   * Return paired devices list.
+   *
+   * @public
+   * @returns {Object[]} sequence of BluetoothDevice
+   */
+  getPairedDevices: function bt__getPairedDevices() {
+    return new Promise((resolve) => {
+      resolve(this._adapter.getPairedDevices());
```

Comment:
I guess you would like to let the interface consistent between v1/v2. But, please leave a note for roll back the interface to return sync call while all of Bluetooth v1 code base is removed in one day.